sale_order_delivery_date/views/stock_picking_views.xml

```diff
@@ -7,8 +7,17 @@
         <field name="inherit_id" ref="stock.view_picking_form"/>
         <field name="arch" type="xml">
             <xpath expr="//field[@name='scheduled_date']" position="after">
-                <field name="delivery_due_date" attrs="{'invisible':[('picking_type_code','!=','outgoing')]}"/>
-                <field name="date_delivered" attrs="{'required':[('picking_type_code','=','outgoing'),('state','=','done')]}" readonly="0"/>
+                <field name="is_customer_outgoing" invisible="1" />
+                <field name="delivery_due_date"
+                    attrs="{'invisible':[('is_customer_outgoing','=',False)]}"
+                    />
+                <field name="date_delivered"
+                    attrs="{
+                        'required':[('picking_type_code','=','outgoing'),('state','=','done')],
```

Comment:
```suggestion
                        'required':[('is_customer_outgoing','=',True),('state','=','done')],
```

Reply:
Thank you, I edit and it worked as expected.
Bump up version too.